src/webgpu/api/operation/rendering/basic.spec.ts

```diff
@@ -129,4 +137,191 @@ g.test('large_draw')
       .combine('indexed', [true, false])
       .combine('indirect', [true, false])
   )
-  .unimplemented();
+  .fn(async t => {
+    const { indexed, indirect } = t.params;
+
+    const BYTES_PER_ROW = 256;
+    const dst = t.device.createBuffer({
+      size: 3 * BYTES_PER_ROW,
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST,
+    });
+
+    const indirectBuffer = t.device.createBuffer({
+      size: 20,
+      usage: GPUBufferUsage.INDIRECT | GPUBufferUsage.COPY_DST,
+    });
+    const writeIndirectParams = (count: number, instanceCount: number) => {
+      const params = new Uint32Array(5);
+      params[0] = count; // Vertex or index count
+      params[1] = instanceCount;
+      params[2] = 0; // First vertex or index
+      params[3] = 0; // First instance (non-indexed) or base vertex (indexed)
+      params[4] = 0; // First instance (indexed)
+      t.device.queue.writeBuffer(indirectBuffer, 0, params, 0, 5);
+    };
+
+    const MILLION = 1024 * 1024;
+    const MAX_INDICES = 16 * MILLION;
+    const indexBuffer = t.device.createBuffer({
+      size: MAX_INDICES * Uint32Array.BYTES_PER_ELEMENT,
+      usage: GPUBufferUsage.INDEX | GPUBufferUsage.COPY_DST,
+    });
+    const indexData = new Uint32Array(MILLION);
+    for (let offset = 0; offset < MAX_INDICES; offset += MILLION) {
+      for (let k = 0; k < MILLION; ++k) {
+        indexData[k] = offset + k;
+      }
+      t.device.queue.writeBuffer(
+        indexBuffer,
+        offset * Uint32Array.BYTES_PER_ELEMENT,
+        indexData,
+        0,
+        MILLION
+      );
+    }
```

Comment:
nit: Rather than a series of writeBuffer calls, I think you could safely do this by specifying `mappedAtCreation: true` in createBuffer, getting a mapped range with `getMappedRange()`, and then writing the test values into that with a (now simpler) for-loop.

Reply:
Aha. Missed this wonderful detail of the spec, that mappedAtCreation can be used without MAP_WRITE or MAP_READ. Nice.